pkg/server/plugin/upstreamauthority/awspca/pca.go

```diff
@@ -61,6 +61,32 @@ type Configuration struct {
 	SupplementalBundlePath  string `hcl:"supplemental_bundle_path" json:"supplemental_bundle_path"`
 }
 
+func NewConfiguration(text string, core *configv1.CoreConfiguration) (config *Configuration, notes []string, err error) {
```

Comment:
Why don't have a `validate` function that has the validation logic only? It doesn't seem like creating a new config is needed when performing a validation. The returned config will always be discarded.
In the same way as this function, the `validate` function could be called from the Configure and Validate functions.

Reply:
The function does not need to be exported.

Reply:
A plugin.Validate gRPC that is independent of the plugin.Configure gRPC would mean that plugin configuration validation would have to be cut-and-pasted twice into the Validate and Configure gRPC calls.  Rather than do that, both calls should defer to one config checking routine (as discussed and accepted by the maintainers).  By pushing that routine into the NewConfiguration func, all new configurations are now centralizing all of the configuration checking.

Reply:
@amartinezfayo The `Configuration` struct across most (if not all) plugins is also exported.  Would you like for me to change that at the same time I change `NewConfiguration`?

Reply:
@amartinezfayo In review with @azdagron a different name for the function was suggested, `newConfig`, which also is not exported.  That method is applied as a parameter to the new functor `pluginconf.Build` which is exported because it lies in a different module.

I'm closing this item as the method in the Plugin is no longer exported.  If you feel I'm closing this in error, please reopen it.

As for only validating the configuration, without creating a new `Configuration` object, such a thing cannot be done because the HCL parsing libraries require a `Configuration` object to parse the HCL out of the text string (now named `hclText`.  While it might be a point to argue where that `Configuration` object is created prior to the HashiCorp parsing of HCL1, there is no way to validate the configuration is HCL1 without parsing it, requiring the prototype, unfilled `Configuration` object.

After that object is filled with values as a side-effect of the parsing, additional validation is performed on the fields post-parsing for many of the plugins.  That validation clearly needs to happen in both the `Configure(...)` and `Validate(...)` gRPC calls, `Configure(...)` to ensure that it is checked before being used, and `Validate(...)` to ensure it is checked before being considered valid.  We cannot validate the values in `validation` logic only, unless we have the `Configure(...)` gRPC call `validate`; but, if we do that, then the return values of a pure-validation-only call won't return a `Configure` object and we will have to build the configuration object twice.

This solution, with the validation returning a `Configuration` object, the list of `notes`, and a possible `error` satisfies both calls from gRPCs `Configure(...)` and `Validate(...)` by letting `Configure(...)` discard the notes and `Validate(...)` discard the Configuration object.